### PR TITLE
Management api

### PR DESCRIPTION
**Pull request by AlexandrosMor**
**Description**

Modify the http client
- Extend CurlClient to support rest api methods

Add management API the following endpoints:
- Get /merchants
- Get /merchants/{merchantId}
- Get /me
- Post /merchants/{merchantId}/apiCredentials/{apiCredentialId}/generateClientKey
- Get /companies
- Get /companies/{companyId}
- Post /merchants/{merchantId}/webhooks/
- Get /companies/{companyId}/webhooks
- Post /merchants/{merchantId}/webhooks/{webhookId}/generateHmac

**Tested scenarios**

Added integration tests

### CODE REVIEW

File: src/Adyen/HttpClient/CurlClient.php [hunk 1]
```diff
@@ -51,97 +52,7 @@ class CurlClient implements ClientInterface
...
-        }
-
-        // result in array or json
-        if ($config->getOutputType() == 'array') {
-            // transform to PHP Array
-            $result = json_decode($result, true);
-        }
-
-        return $result;
+        return $this->requestHttp($service, $requestUrl, $params, 'post', $requestOptions);
```
Comment (acampos1916) [inline src/Adyen/HttpClient/CurlClient.php:55]: `$requestOptions` doesn't seem to be an arg in the ClientInterface function?

File: src/Adyen/Service/AbstractResource.php [hunk 2]
```diff
@@ -216,4 +204,47 @@ private function isBase64Encoded($data)
...
+     * @param $url
+     * @param $method
+     * @return mixed
+     * @throws AdyenException
+     */
+    public function requestHttp($params, $url, $method)
+    {
+        // check if rest api method has a value
+        if (!$method) {
+            $msg = 'The rest api method is empty';
```
Comment (acampos1916) [inline src/Adyen/Service/AbstractResource.php:237]:
```suggestion
            $msg = 'The REST API method is empty';
```

File: tests/Integration/ManagementTest.php [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    public function testCreateMerchantWebhooks()
+    {
+        if (empty($this->settings['merchantAccount']) ||
+            $this->settings['merchantAccount'] == self::YOUR_MERCHANT_ACCOUNT) {
+            $this->skipTest(self::SKIP_TEST_MESSAGE);
+            return null;
+        }
+        $params = array(
+            "type" => "standard",
+            "url" => "https://magento2-devx.ataberkylmz.com/",
```
Comment (acampos1916) [inline tests/Integration/ManagementTest.php:143]: Can we supply a non working URL here instead or does this service also test the webhook being accepted?

Comment (AlexandrosMor) [inline tests/Integration/ManagementTest.php:143]:
```suggestion
            "url" => "https://testwebsite.com/",
```

File: src/Adyen/Service/ResourceModel/Management/MerchantClientKey.php [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+     * @param $apiCredentialId
+     * @return mixed
+     * @throws \Adyen\AdyenException
+     */
+    public function create($params, $merchantId, $apiCredentialId)
+    {
+        $url = $this->managementEndpoint . "/merchants/" . $merchantId . "/apiCredentials/"
+            . $apiCredentialId . "/generateClientKey";
+        return $this->requestHttp($params, $url, 'post');
+    }
```
Comment (michaelpaul) [inline src/Adyen/Service/ResourceModel/Management/MerchantClientKey.php:42]: I think path parameters should come before body params for this and other POST requests.: 
```suggestion
    public function create($merchantId, $apiCredentialId, $params)
    {
        $url = $this->managementEndpoint . "/merchants/" . $merchantId . "/apiCredentials/"
            . $apiCredentialId . "/generateClientKey";
        return $this->requestHttp($url, 'post', $params);
    }
```

**Review by michaelpaul — CHANGES_REQUESTED** [review]
It's looking good! I'm just leaving a few suggestions.

File: src/Adyen/Service/Management.php [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+     */
+    public function __construct(\Adyen\Client $client)
+    {
+        parent::__construct($client);
+        $this->merchantAccount = new MerchantAccount($this);
+        $this->me = new Me($this);
+        $this->companyAccount = new CompanyAccount($this);
+        $this->merchantClientKey = new MerchantClientKey($this);
+        $this->merchantWebhooks = new MerchantWebhooks($this);
+        $this->companyWebhooks = new CompanyWebhooks($this);
```
[the remark this answers is not in the document]
Comment (AlexandrosMor) [reply src/Adyen/Service/Management.php:59]: I do not see why to add the `null` as parameter as parameter, If we add the null is only for the endpoint checking  [here](https://github.com/Adyen/adyen-php-api-library/blob/PW-6124/src/Adyen/Service/AbstractResource.php#L43)

**Review by michaelpaul — APPROVED** [review]

**Review by acampos1916 — APPROVED** [review]

**Pull request merged**